### ISSUE TITLE
fix: Person Re-ID model fails to load due to classifier shape mismatch

### ISSUE THREAD

**Issue by mikesvoboda** [mikesvoboda/nemotron-v3-home-security-intelligence#4508]
## Problem

The OSNet-x0.25 person Re-ID model fails to load during warmup in the `ai-enrichment-light` container due to a shape mismatch in the classifier layer.

## Error

```
ERROR - Failed to load person Re-ID: Error(s) in loading state_dict for OSNet:
  size mismatch for classifier.weight: copying a param with shape torch.Size([1000, 512]) 
    from checkpoint, the shape in current model is torch.Size([1, 512]).
  size mismatch for classifier.bias: copying a param with shape torch.Size([1000]) 
    from checkpoint, the shape in current model is torch.Size([1]).
```

## Root Cause

The checkpoint was trained with 1000 output classes (likely ImageNet pretrained), but the model is being instantiated with `num_classes=1`. For Re-ID, we typically use the model as a feature extractor (512-dim embeddings) and don't need the classifier head at all.

## Location

* **Container**: `ai-enrichment-light`
* **Service**: Person Re-ID / `backend/services/reid_service.py`
* **Model**: OSNet-x0.25

### Option 1: Load with strict=False (Recommended)

```python
# Load weights, ignoring mismatched classifier
state_dict = torch.load(checkpoint_path)
model.load_state_dict(state_dict, strict=False)
# The classifier will use random weights, but we don't use it anyway
```

### Option 2: Remove classifier keys before loading

```python
state_dict = torch.load(checkpoint_path)
# Remove classifier weights from state dict
state_dict = {k: v for k, v in state_dict.items() if not k.startswith('classifier')}
model.load_state_dict(state_dict, strict=False)
```

### Option 3: Match checkpoint num_classes

```python
# Initialize model with same num_classes as checkpoint
model = OSNet(num_classes=1000)
model.load_state_dict(torch.load(checkpoint_path))
# Then replace or ignore classifier for inference
model.classifier = nn.Identity()  # or just don't use it
```

## Acceptance Criteria

- [ ] Person Re-ID model loads successfully without errors
- [ ] Re-ID warmup completes in `ai-enrichment-light` container
- [ ] Person tracking works across cameras (embeddings extracted correctly)
- [ ] Add unit test for model loading with different num_classes

## Testing

```bash
# Check Re-ID service tests
uv run pytest backend/tests/unit/services/test_reid_service.py -v

# Integration test for person tracking
uv run pytest backend/tests/integration/ -k "reid" -v

# Manual verification
podman logs nemo3-ai-enrichment-light-1 2>&1 | grep -i "re-id"
```

## Impact

* **Severity**: Critical
* **Affected Service**: Person re-identification across cameras
* **User Impact**: Cannot track individuals across different camera views
